### PR TITLE
move source files to a new Source directory and decrease build times

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
The diff for some files seem to show that the whole file has changed, so I'll add a few notes here:

- I've moved the source files (.h, .cpp, .ui) to the new `Source` directory.

- I reduced the build times by forward declaring many classes and by adding a pch.h file, which includes common files that were recompiled many times.

- I removed the `CockHeroine.pro.user` file which was added to `.gitignore` but was never deleted after

File: CockHeroine.pro [hunk 1]
```diff
@@ -11,200 +11,201 @@ greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 TARGET = CockHeroine
 TEMPLATE = app
 
+PRECOMPILED_HEADER = Source/pch.h
```
Comment (memmaa) [inline CockHeroine.pro:14]: qmake needs this:
```suggestion
PRECOMPILED_HEADER = Source/pch.h

INCLUDEPATH += Source/
```

**Comment by memmaa** [comment]
> The diff for some files seem to show that the whole file has changed

My guess is it's because you've changed the line endings from unix to Windows. You'd think github would handle that better (and I guess I never set up a .gitattributes. I've now added one)...
No worries - Meld/WinMerge are up to the task. :-)

>  I reduced the build times by forward declaring many classes and by adding a pch.h file, which includes common files that were recompiled many times.

Thanks! I appreciate your time and effort in tidying up the project. I only hope you get some use out of it in return!  
Did you use a tool to automate that?

>  I removed the `CockHeroine.pro.user` file which was added to `.gitignore` but was never deleted after

👍 

I gave  it a run through Qt Creator and the new Source folder needs adding as an include path. I've added a comment

**Comment by memmaa** [comment]
Although closed, the changes in this PR have made it into master via another means. Even with autocrlf, git still didni't accept the changes well, it even started detecting files with no changes (same md5sum) as changed. A `git add -uv .` did the trick in the end.

**Pull request closed without merging**